### ISSUE TITLE
auth.login.CredentialNotFoundException: Unable to find a valid credential with provided parameters

### ISSUE THREAD

**Issue by bloudman** [jenkinsci/azure-keyvault-plugin#21]
### Your checklist for this issue

- [ x ] Jenkins version

- [ x ] Plugin version

- [ x ] OS

### Description

Please describe your issue here.

I'm in trouble with the latest version of this plugin on the latest version of jenkins.

I tried this code: 
```groovy
                def secrets = [
                  [ secretType: 'Secret', name: 'databasePassword', version: '', envVariable: 'DB_PASSWORD' ],
                  [ secretType: 'Secret', name: 'databaseUsername', version: '', envVariable: 'DB_USERNAME' ]
                ]
                withAzureKeyvault(azureKeyVaultSecrets: secrets, 
                    keyVaultURLOverride: 'https://keyvault.vault.azure.net/',
                    credentialIDOverride: 'mySP'
                ) {
                    ansiblePlaybook installation: 'ansible',
                    playbook: 'webapp.yml'
                  }

```
but it does not work. I can only use the plugin with the UI.
I have this error:

```
org.jenkinsci.plugins.azurekeyvaultplugin.AzureKeyVaultException: No key vault url configured, set one globally or in the build wrap step
	at org.jenkinsci.plugins.azurekeyvaultplugin.AzureKeyVaultBuildWrapper.getKeyVaultURL(AzureKeyVaultBuildWrapper.java:148)
	at org.jenkinsci.plugins.azurekeyvaultplugin.AzureKeyVaultBuildWrapper.setUp(AzureKeyVaultBuildWrapper.java:286)
	at org.jenkinsci.plugins.workflow.steps.CoreWrapperStep$Execution2.doStart(CoreWrapperStep.java:97)
	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution.lambda$run$0(GeneralNonBlockingStepExecution.java:77)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```
Is it possible to pass the keyvault url and the service principal in argument ?

Thanks

**Comment by timja** [comment]
Hi 
@bloudman apologies for the long delay, apparently I wasn't watching this repo, (fixed).

I've tested this on the master branch, (I did some refactoring in this area recently)

It worked for me with:
```
node {
  withAzureKeyvault(
      credentialIDOverride: 'sp',
      azureKeyVaultSecrets: [[envVariable: 'CRED', name: 'cred', secretType: 'Secret'], [envVariable: 'CERT_LOCATION', name: 'cert', secretType: 'Certificate']]) {
    sh "env | grep SANDBOX"
    sh "openssl pkcs12 -in $CERT_LOCATION  -nodes -password 'pass:' -out keyvault.pem"
  }   
}
```

Can you please re-test on

**Comment by timja** [comment]
This should be fixed in 1.3
